### PR TITLE
feat(ui): Add component for selecting GIFs.

### PR DESCRIPTION
**Pull request by NoLogicAlan**
## Please make sure to check the following tasks before opening and submitting a PR

* [x] I understand and have followed the [contribution guide](https://github.com/revoltchat/revolt/discussions/282)
* [x] I have tested my changes locally and they are working as intended
* [x] These changes do not have any notable side effects on other Revolt projects

### CODE REVIEW

**Comment by NoLogicAlan** [comment]
I have read the CLA Document and I hereby sign the CLA

File: components/ui/components/floating/CompositionPicker.tsx [hunk 1]
```diff
@@ -22,22 +22,42 @@ const Base = styled(Column)`
 /**
  * Container element for the picker
  */
+const SearchBar = styled(Input)`
+  border-radius: 3px;
```
Comment (insertish) [inline components/ui/components/floating/CompositionPicker.tsx:26]: Not determined by theme

File: components/ui/components/floating/CompositionPicker.tsx [hunk 2]
```diff
@@ -22,22 +22,42 @@ const Base = styled(Column)`
...
+  overflow: hidden;
+  border: 2px solid #ccc;
+  padding: 12px;
```
Comment (insertish) [inline components/ui/components/floating/CompositionPicker.tsx:29]: Not determined by theme

File: components/ui/components/floating/CompositionPicker.tsx [hunk 3]
```diff
@@ -22,22 +22,42 @@ const Base = styled(Column)`
...
 const Container = styled("div", "Picker")`
   flex-grow: 1;
   display: flex;
   height: 400px;
   flex-direction: column;
-
+‎
   color: white;
-  background-color: black;
+  background-color: #242424;
```
Comment (insertish) [inline components/ui/components/floating/CompositionPicker.tsx:39]: Not determined by theme

File: components/ui/components/floating/CompositionPicker.tsx [hunk 4]
```diff
@@ -22,22 +22,42 @@ const Base = styled(Column)`
...
   border-radius: ${(props) => props.theme!.borderRadius.md};
+  padding: 10px;
```
Comment (insertish) [inline components/ui/components/floating/CompositionPicker.tsx:41]: Not determined by theme

File: components/ui/components/floating/CompositionPicker.tsx [hunk 5]
```diff
@@ -22,22 +22,42 @@ const Base = styled(Column)`
...
+  box-shadow: 0 0 10px rgba(0, 0, 0, 0.25);
 `;
 
 const GifList = styled(ScrollContainer)`
   display: grid;
-  gap: 10px;
-  grid-template-columns: repeat(auto-fill, minmax(160px, 1fr));
-  grid-template-rows: masonry;
+  grid-template-columns: repeat(2, 1fr);
+  gap: 6px;
```
Comment (insertish) [inline components/ui/components/floating/CompositionPicker.tsx:49]: Not determined by theme

File: components/ui/components/floating/CompositionPicker.tsx [hunk 6]
```diff
@@ -22,22 +22,42 @@ const Base = styled(Column)`
...
+  grid-auto-rows: 200px; /* Set a fixed height */
+  overflow: auto;
+  padding: 4px;
```
Comment (insertish) [inline components/ui/components/floating/CompositionPicker.tsx:52]: Not determined by theme

File: components/ui/components/floating/CompositionPicker.tsx [hunk 7]
```diff
@@ -160,8 +180,8 @@ export function CompositionPicker(props: Props) {
                 role="tooltip"
               >
                 <Container>
-                  <Input
-                    placeholder="search for gifs :)"
+                  <SearchBar
+                    placeholder="Type to search..."
```
Comment (insertish) [inline components/ui/components/floating/CompositionPicker.tsx:184]: i18n

**Review by insertish — COMMENTED** [review]
Forgot to actually commit these comments to a review

**Comment by insertish** [comment]
The whole GIF picker will be re-done with the masonry layout from GIFBox, I would probably hold off on making any changes to it for now.

**Comment by insertish** [comment]
Closing due to inactivity

**Pull request closed without merging**